jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/DeploymentScanner.java

```diff
@@ -1185,25 +1188,26 @@ public void setContainerScanJarPattern(String pattern)
         }
 
         /**
-         * The name of the class that this environment uses to create {@link ContextHandler}
-         * instances (supports a class that implements {@code java.util.function.Supplier<Handler>} as well).
+         * The name of the class that this environment uses to create {@link Deployable.ClassLoaderFactory}
+         * instances.
          *
          * <p>
-         * This is the class used to create a ContextHandler for the environment before
-         * any XML files are loaded to configure the context.
+         * This is the class used to create a ClassLoader for the environment before
+         * any XML files are loaded to create/configure the {@link ContextHandler}.
          * </p>
          *
-         * @param classname the classname for this environment's context deployable.
-         * @see StandardContextHandlerFactory#CONTEXT_HANDLER_CLASS_ATTRIBUTE
+         * @param classname the classname for this environment's classloader factory.
+         * @see StandardContextHandlerFactory#CLASSLOADER_FACTORY_ATTRIBUTE
+         * @see org.eclipse.jetty.server.Deployable.ClassLoaderFactory
          */
-        public void setContextHandlerClass(String classname)
+        public void setClassLoaderFactoryClass(String classname)
```

Comment:
Rename to `setClassLoaderFactoryClassName(String)` in case we want later to have a version that takes a `Class<?>` parameter.